exercises/transpose/examples/success-standard/src/Transpose.hs

```diff
@@ -0,0 +1,22 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Transpose (transpose) where
+
+import Data.Maybe (isNothing)
+
+transpose :: [String] -> [String]
+transpose [] = []
+transpose rows =
+  map toString $ foldr 
+    (zipWith (:) . padRight . map Just) 
+    (replicate longestLength []) 
+    rows
+  where 
+    longestLength = maximum $ map length rows
+    padRight line = line ++ replicate (longestLength - length line) Nothing
+    toString line = concatMap (\case Nothing -> " "
```

Comment:
It looks like the lambda always results in a single-character string), in which case I think `map` would make work just as well as `concatMap` and be clearer. Let me know if `map` can work, OR if I misunderstood the code and it's actually not possible.

Reply:
@petertseng this piece of the code confused me too when i was writing it because with just `map`, it ends up being `[[[Char]]]`. I used `concatMap` because I needed to use `concat` and `map` to remove that nested layer.

Reply:
Hold on, I don't think I understand this explanation and where `[[[Char]]]` comes from. Can you show me some code and explain what you mean and what compiler error occurs?

What I am saying is that these two pices of code are equivalent:

```haskell
xs = [1, 2, 3]
exprA = concatMap (\x -> [x + 1]) xs
exprB = map (\x -> x + 1) xs
```

I think the code as written in this PR is equivalent to exprA, and I think it should be exprB.